### ISSUE TITLE
Bug: <short description of the issue>

### ISSUE THREAD

**Issue by motss** [typescript-eslint/typescript-eslint#9015]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

```ts
import url from 'node:url';

import eslint from '@eslint/js';
import tseslint from 'typescript-eslint';
import { FlatCompat } from '@eslint/eslintrc';

const __dirname = url.fileURLToPath(new URL('.', import.meta.url));
const _compat = new FlatCompat({ baseDirectory: __dirname });

export default tseslint.config(
  /**
   * register all of the plugins up-front
   */
  {
    plugins: {
      '@typescript-eslint': tseslint.plugin,
    },
  },
  
  /**
   * extends
   */
  eslint.configs.recommended,
  ...tseslint.configs.strictTypeChecked,

  /**
   * base config
   */
  {
    languageOptions: {
      parserOptions: {
        project: ['./tsconfig.json'],
        tsconfigRootDir: __dirname,
        warnOnUnsupportedTypeScriptVersion: true,
      },
    },
    rules: {
      '@typescript-eslint/no-unused-vars': [
        'error',
        {
          // reportUsedIgnorePattern: true,
          args: 'all',
          argsIgnorePattern: '^_',
          destructuredArrayIgnorePattern: '^_',
          caughtErrors: 'all',
          varsIgnorePattern: '^_',
        },
      ],
    },
  },

  {
    files: ['eslint.config.{js,cjs,mjs}'],
    rules: {
      'import/no-default-export': 'off',
    },
  }
);
```

ESLint reports linting errors on its own packages in `eslint.config.mjs`.

```sh
/eslint.config.mjs
   7:7   error  Unsafe assignment of an `any` value                                                @typescript-eslint/no-unsafe-assignment
   7:19  error  Unsafe call of an `any` typed value                                                @typescript-eslint/no-unsafe-call
   7:23  error  Unsafe member access .fileURLToPath on an `any` value                              @typescript-eslint/no-unsafe-member-access
   8:7   error  Unsafe assignment of an `any` value                                                @typescript-eslint/no-unsafe-assignment
   8:17  error  Unsafe construction of an any type value                                           @typescript-eslint/no-unsafe-call
   8:34  error  Unsafe assignment of an `any` value                                                @typescript-eslint/no-unsafe-assignment
  30:3   error  Unsafe argument of type `any` assigned to a parameter of type `ConfigWithExtends`  @typescript-eslint/no-unsafe-argument
  30:10  error  Unsafe member access .configs on an `any` value                                    @typescript-eslint/no-unsafe-member-access
  40:9   error  Unsafe assignment of an `any` value                                                @typescript-eslint/no-unsafe-assignment

✖ 9 problems (9 errors, 0 warnings)
```

### Reproduction Repository Link

https://stackblitz.com/edit/vitest-dev-vitest-si258e

### Repro Steps

1) Go to https://stackblitz.com/edit/vitest-dev-vitest-si258e
2) Wait for the dependencies installation and you should see the linting errors
3) If Step 2 fails, manually run `pnpm install && pnpm lint`

### Versions

```ts
  "@eslint/eslintrc": "^3.0.2",
  "@eslint/js": "^9.1.1",
  "eslint": "^9.1.1",
  "typescript-eslint": "^7.7.1"
```

**Comment by motss** [comment]
>  It's untyped JS and you haven't installed any @types/ packages to provide types!
I see. The official docs recommends that but never mention about the additional type packages are required.

**Comment by bradzacher** [comment]
in general you would avoid linting your config files because they are pure JS files and aren't likely to be typed.
